plugins/pipelines/app/decorators/stage_decorator.rb

```diff
@@ -2,5 +2,10 @@
   prepend SamsonPipelines::StageConcern
   serialize :next_stage_ids, Array
 
-  validate :valid_pipeline?
+  validate :valid_pipeline?, if: :next_stage_ids_changed?
+
+  # duplicate call from models/stage.rb, but this is needed
+  # to load the soft delete methods
```

Comment:
hmm decorators should be loaded after the base class is loaded I guess ... but that might be tricky :(